ajax_select/fields.py

```diff
@@ -365,12 +366,14 @@ def __init__(self, channel, *args, **kwargs):
         widget_kwargs.update(kwargs.pop('widget_options', {}))
         if 'attrs' in kwargs:
             widget_kwargs['attrs'] = kwargs.pop('attrs')
-        widget = AutoCompleteWidget(channel, **widget_kwargs)
+        widget = AutoCompleteWidget(**widget_kwargs)
 
-        defaults = {'max_length': 255, 'widget': widget}
-        defaults.update(kwargs)
+        default_kwargs = {'max_length': 255}
+        default_kwargs.update(kwargs)
+        kwargs = default_kwargs
```

Comment:
this is very confusing code. it is best not to juggle variables like this.

the main fix you are proposing is just to set widget explicitly and not to allow the kwargs to override that.  the rest of the changes are not needed